metricflow/cli/main.py

```diff
@@ -721,5 +733,181 @@ def validate_configs(cfg: CLIContext, dw_timeout: Optional[int] = None, skip_dw:
         _data_warehouse_validations_runner(dw_validator=dw_validator, model=user_model, timeout=dw_timeout)
 
 
+def _get_spin_context_manager_factory(text: str) -> Callable[[], ContextManager[None]]:
+    @contextlib.contextmanager
+    def _context_manager() -> Iterator[None]:
+        start_ms = int(time.perf_counter() * 1000)
+        spinner = Halo(text=text, spinner="dots")
+        spinner.start()
+        yield
+        end_ms = int(time.perf_counter() * 1000)
+        total_ms = end_ms - start_ms
+        spinner.succeed(text=(click.style(f"{total_ms}ms ", fg="yellow") + text))
+
+    return _context_manager
+
+
+class CLIInferenceProgressReporter(InferenceProgressReporter):
+    """Writes inference progress to stdout as pretty output."""
+
+    @staticmethod
+    @contextlib.contextmanager
+    def warehouse() -> Iterator[None]:  # noqa: D
+        yield
+
+    @staticmethod
+    @contextlib.contextmanager
+    def table(table: SqlTable, index: int, total: int) -> Iterator[None]:  # noqa: D
+        with _get_spin_context_manager_factory(f"🔍 Querying `{table.sql}` ({index + 1} out of {total})")():
+            yield
+
+    rules = staticmethod(_get_spin_context_manager_factory("🤔 Processing inference rules"))  # type: ignore
```

Comment:
I'm assuming the `type:  ignore` is needed due to the style of staticmethod construction here? If so, we should definition just make it through the decorator as normal. The type ignore is not great and I think the style of `staticmethod(...)` is considered unpythonic

Reply:
Will do.